docs/eip-draft_bls_remote_signer_api.md

```diff
@@ -0,0 +1,418 @@
+
+## Specification
+
+### `GET /upcheck`
+
+_**Responses**_
+
+Success | <br>
+--- | ---
+Code | `200`
+Content | `{"status": "OK"}`
+
+---
+
+### `GET /publicKeys`
+
+_**Responses**_
+
+Success | <br>
+--- | ---
+Code | `200`
+Content | `{"public_keys": "[public_key_hex_string_without_0x]"}`
+
+_or_
+
+Error | <br>
+--- | ---
+Code | `404`
+Content | `{"error": "No keys found in storage."}`
+
+_or_
+
+Error | <br>
+--- | ---
+Code |  `500`
+Content|  `{"error": "<Server Error Message>"}`
+
+---
+
+### `POST /sign/:public_key`
+
+URL Parameter | <br>
+--- | ---
+`:public_key` | `public_key_hex_string_without_0x`
+
+_**Request**_
+
+JSON Body | <br> | <br>
+--- | --- | ---
+`signingRoot` | **Required** | A string representation of the hexadecimal value to be signed
+<br> | Optional | Any other field will be ignored by the signer
+
+_**Responses**_
+
+Success | <br>
+--- | ---
+Code |  `200`
+Content | `{"signature": "<signature_hex_string>"}`
+
+_or_
+
+Error | <br>
+--- | ---
+Code |  `400`
+Content | `{"error": "<Bad Request Error Message>"}`
+
+_or_
+
+Error | <br>
+--- | ---
+Code |  `404`
+Content | `{"error": "Key not found: <public_key_hex_string_without_0x>"}`
+
+_or_
+
+Error | <br>
+--- | ---
+Code |  `500`
+Content | `{"error": "<Server Error Message>"}`
+
+---
+
+## Rationale
+
+### URL parameter `:public-key` without `0x`
+
+By relaxing the constraint of using an hexadecimal, clients can opt-in to using custom ids for their private keys, and perform the adequate retrieving and matching with the secret key within the signer. This is, in the measure the remote signer implementer allows for this feature.
+
+### UNIX philosophy: Simple API
+
+This API specification contains only three methods: one for **status**, one for **listing the available keys**, and one to **produce a signature**. There are no methods for authentication, pre-image validation, key management, nor transport encryption.
+
+The following subsections discuss aspects to be considered by the client implementers relative to these subjects.
+
+#### Authentication
+
+Can be accomplished by either prepending to the API, or adding a middleware to the client implementation the validation of an HTTP Request Header.
+
+There are several ways to negotiate and issue a valid token to authenticate the communication between the validator client and the remote signer, each of them with potential drawbacks (e.g replay attacks, challenges in distributing the token to the validator client, etc.). In general, any method of authentication must be combined with transport encryption to be effective.
+
+The operator can also implement network Access Control Lists (ACLs) between the validator client's network and the remote signer's network, reducing the attack surface by requiring a potential attacker to be positioned in the same network as the validator client.
+
+#### Pre-image validation
```

Comment:
I guess there's an argument for making the pre-image mandatory, so that the signer can verify the validity of the messages being signed. I think I'm happy with leaving it out of the EIP for now, but I suspect other people might think differently.

Reply:
Oh yes. This is a controversial topic :hot_pepper: :hot_pepper: :hot_pepper: 

My vote is to keep this feature **OUT** of the specification, as this is the road to feature creep. For example, if we do slashing validation, we want to maintain the working and maintainance of a slashing database, as opposed to let everybody choose the proxy / plugin that better suits their needs, or even none at all (if you are the YOLO kind of person).

I hold the same suspicion you have, people will ask for it. Let's see how it goes, and adapt accordingly.

Reply:
Sounds good!